src/main/java/com/github/manolo8/darkbot/core/api/GameAPIImpl.java

```diff
@@ -279,6 +283,42 @@ public void readMemory(long address, byte[] buffer, int length) {
         memory.readBytes(address, buffer, length);
     }
 
+    @Override
+    public DataReader readData(long address, int length) throws RuntimeException {
+        if (length <= 0 || length > DataReader.MAX_CHUNK_SIZE)
+            throw new ArrayIndexOutOfBoundsException("Length is <= 0 or exceeds max chunk size: " + DataReader.MAX_CHUNK_SIZE);
+
+        for (int i = 0; i < dataReaders.length; i++) {
+            AbstractDataReader reader = (AbstractDataReader) dataReaders[i];
+            if (reader == null) reader = (AbstractDataReader) (dataReaders[i] = createReader(i));
```

Comment:
```suggestion
            DataReader reader = dataReaders[i];
            if (reader == null) reader = (dataReaders[i] = createReader(i));
```
Any reason this can't rely only on the interface, and not the implementation?